File: backend/user.service/user/src/API/Controllers/User.Controller.cs
using Azure.Core;
using Domain.Models;
using Microsoft.AspNetCore.Mvc;
using user.src.API.Models.User;
using user.src.Domain.Services;

namespace user.src.API.Controllers
{
	[Route("api")]
	[ApiController]
	public class UserController : ControllerBase
	{
		private readonly UserService userService;
		public UserController(UserService user)
		{
			userService = user;
		}
		[HttpGet("fetch-list-user")]
		public async Task<IActionResult> FetchListUser()
		{
			try
			{
				var Response = await userService.FetchListUser();
				return Ok(Response);
			}
			catch (Exception ex)
			{
				return BadRequest(new { message = ex.Message });
			}
		}
		[HttpGet("get-user/{idUser}")]
		public async Task<IActionResult> GetUserById([FromRoute] Guid idUser)
		{

			try
			{
				var Response = await userService.GetUserAsync(idUser);
				return Ok(Response);
			}
			catch (Exception ex)
			{
				Console.WriteLine(ex);
				return BadRequest(new { message = ex.Message });
			}
		}
		[HttpPut("user/{id}")]
		public async Task<IActionResult> EditUserById([FromRoute] Guid id, [FromBody] UpdateUserRequest req)
		{
			try
			{
				//get user by id
				var user = await userService.GetUserAsync(id);
				if (user == null)
					return NotFound("Người dùng không tồn tại!");
				Console.WriteLine(user.BirthDay);
				//update user
				user.NameUser = req.NameUser ?? user.NameUser;
				user.BirthDay = req.date ?? user.BirthDay;
				//save user 
				var result = await userService.EditUserAsync(user);
				if (!result) return BadRequest("Cập nhật thông tin thất bại!");
				return Ok("Cập nhật thông tin thành công!");
			}
			catch (Exception ex)
			{
				return BadRequest(new { message = ex.Message });
			}
		}

	}
}

File: backend/Api-Gateway/src/controllers/user.service/user.controller.js
import axios from 'axios';
import config from '../../config.js';
import { HTTP_STATUS } from '../../common/http-status.common.js';

export const userController = {
  //Fetch list user
  fetchListUser: async (req, res) => {
    const result = await axios.get(
      `${process.env.USER_SERVICE_URL}/fetch-list-user`,
    );
    if (!result)
      return res.status(HTTP_STATUS.BAD_REQUEST).json({
        message: 'Tải dữ liệu thất bại!',
        sucess: false,
      });
    return res.status(HTTP_STATUS.OK).json({
      message: 'Tải dữ liệu thành công!',
      sucess: true,
      data: result.data,
    });
  },
  //Get user by id
  getUserById: async (req, res) => {
    const { idUser } = req.params;
    const result = await axios.get(
      `${process.env.USER_SERVICE_URL}/get-user/${idUser}`,
    );
    if (!result)
      return res.status(HTTP_STATUS.BAD_REQUEST).json({
        message: 'Không thể tải dữ liệu người dùng!',
        sucess: false,
      });
    return res.status(HTTP_STATUS.OK).json({
      message: 'Tải dữ liệu người dùng thành công!',
      sucess: true,
      data: result.data,
    });
  },
  //update user by id
  updateUserById: async (req, res) => {
    const { id } = req.params;
    const data = req.body;
    const result = await axios.put(
      `${process.env.USER_SERVICE_URL}/user/${id}`,
      data,
      config,
    );
    if (!result)
      return res.status(HTTP_STATUS.BAD_REQUEST).json({
        message: 'Cập nhật thông tin thất bại!',
        sucess: false,
      });
    return res.status(HTTP_STATUS.OK).json({
      message: 'Cập nhật thông tin thành công!',
      sucess: true,
    });
  },
};